packages/arb-provider-ethers/src/lib/byte_serialize_params.ts

```diff
@@ -0,0 +1,100 @@
+import { isAddress as _isAddress } from '@ethersproject/address'
+import { concat, hexZeroPad } from '@ethersproject/bytes'
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { getAddressIndex } from './l2bridge'
+import { JsonRpcProvider } from 'ethers/providers'
+
+type PrimativeType = string | number | boolean | BigNumber
+type PrimativeOrPrimativeArray = PrimativeType | PrimativeType[]
+type BytesNumber = 1 | 4 | 8 | 16 | 32
+
+// to use:
+// const mySerializeParamsFunction = initSerializeAndLookUpIndices("myAggregatorUrl")
+export const initSerializeAndLookUpIndices = (arbAggregatorUrl: string) => {
```

Comment:
Also what's the "init" part of this function? Seems a bit confusing

Reply:
...yeah, the idea is to "initialize" (maybe "createSerializer" would have been a better name once) the function w/ the aggregator-url (or aggregator) once, and then have the returned function just store it as a closure; it's a bit of a roundabout way to avoid to declare a class. Admittedly, looking w/ fresh eyes, the pattern does feel a bit convoluted